# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1259957897 18000
#      Fri Dec 04 15:18:17 2009 -0500
# Node ID c1210502785ba5bcacfe0df00315506c1114248a
# Parent  3dbe6e920f768e1fa275c319b88707099f9a35fa
provide OCTAVE_API_VERSION_NUMBER

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,14 @@
+2009-12-04  John W. Eaton  <jwe@octave.org>
+
+	* configure.ac (OCTAVE_API_VERSION_NUMBER): New variable.
+	(OCTAVE_API_VERSION): Define using OCTAVE_API_VERSION_NUMBER.
+	* common.mk (OCTAVE_API_VERSION_NUMBER): New variable.
+
 2009-12-04  John W. Eaton  <jwe@octave.org>
 
 	* configure.ac: Check $have_x = yes, not just $have_x.
 	From Carmine Paolino <earcar@gmail.com>.
 
 2009-12-04  John W. Eaton  <jwe@octave.org>
 
 	* configure.ac: Require auotconf 2.62.
diff --git a/common.mk b/common.mk
--- a/common.mk
+++ b/common.mk
@@ -14,16 +14,17 @@ export PERL
 
 PYTHON = @PYTHON@
 
 GNUPLOT = @GNUPLOT@
 
 DESKTOP_FILE_INSTALL = @DESKTOP_FILE_INSTALL@
 
 OCTAVE_VERSION = @OCTAVE_VERSION@
+OCTAVE_API_VERSION_NUMBER = @OCTAVE_API_VERSION_NUMBER@
 OCTAVE_API_VERSION = @OCTAVE_API_VERSION@
 OCTAVE_RELEASE_DATE = @OCTAVE_RELEASE_DATE@
 OCTAVE_COPYRIGHT = @OCTAVE_COPYRIGHT@
 
 version = $(OCTAVE_VERSION)
 api_version = $(OCTAVE_API_VERSION)
 
 ## AM_LIBTOOLFLAGS = --silent
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -28,20 +28,22 @@ dnl               2009 John W. Eaton
 
 EXTERN_CFLAGS="$CFLAGS"
 EXTERN_CXXFLAGS="$CXXFLAGS"
 
 AC_INIT([GNU Octave], [3.3.50+], [bug@octave.org], [octave], [http://www.octave.org])
 
 dnl PACKAGE_VERSION is set by the AC_INIT VERSION arg
 OCTAVE_VERSION="$PACKAGE_VERSION"
-OCTAVE_API_VERSION="api-v38+"
+OCTAVE_API_VERSION_NUMBER="38"
+OCTAVE_API_VERSION="api-v$OCTAVE_API_VERSION_NUMBER+"
 OCTAVE_RELEASE_DATE="2009-09-12"
 OCTAVE_COPYRIGHT="Copyright (C) 2009 John W. Eaton and others."
 AC_SUBST(OCTAVE_VERSION)
+AC_SUBST(OCTAVE_API_VERSION_NUMBER)
 AC_SUBST(OCTAVE_API_VERSION)
 AC_SUBST(OCTAVE_RELEASE_DATE)
 AC_SUBST(OCTAVE_COPYRIGHT)
 
 AM_INIT_AUTOMAKE([1.11 tar-ustar])
 AC_REVISION($Revision: 1.603 $)
 AC_PREREQ([2.62])
 AC_CONFIG_SRCDIR([src/octave.cc])
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-04  John W. Eaton  <jwe@octave.org>
+
+	* version.h.in (OCTAVE_API_VERSION_NUMBER): New macro.
+	* Makefile.am (version.h): Substitute OCTAVE_API_VERSION_NUMBER.
+
 2009-12-04  John W. Eaton  <jwe@octave.org>
 
 	* Makefile.am (mkdefs): New arguments, srcdir and file.
 	Generate output here.
 	(.cc.df): Not here.
 
 2009-12-04  John W. Eaton  <jwe@octave.org>
 
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -595,16 +595,17 @@ builtins.cc: $(DEF_FILES) mkbuiltins
 
 mxarray.h: mxarray.h.in Makefile
 	$(SED) < $< \
 	  -e "s|%OCTAVE_IDX_TYPE%|${OCTAVE_IDX_TYPE}|" > $@-t
 	mv $@-t $@
 
 version.h: version.h.in Makefile
 	$(SED) < $< \
+	  -e "s|%OCTAVE_API_VERSION_NUMBER%|${OCTAVE_API_VERSION_NUMBER}|" \
 	  -e "s|%OCTAVE_API_VERSION%|\"${OCTAVE_API_VERSION}\"|" \
 	  -e "s|%OCTAVE_COPYRIGHT%|\"${OCTAVE_COPYRIGHT}\"|" \
 	  -e "s|%OCTAVE_RELEASE_DATE%|\"${OCTAVE_RELEASE_DATE}\"|" \
 	  -e "s|%OCTAVE_VERSION%|\"${OCTAVE_VERSION}\"|" > $@-t
 	mv $@-t $@
 
 graphics.h: graphics.h.in genprops.awk Makefile
 	$(AWK) -f $(srcdir)/genprops.awk $< > $@-t
diff --git a/src/version.h.in b/src/version.h.in
--- a/src/version.h.in
+++ b/src/version.h.in
@@ -22,16 +22,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #if !defined (octave_version_h)
 #define octave_version_h 1
 
 #define OCTAVE_VERSION %OCTAVE_VERSION%
 
+#define OCTAVE_API_VERSION_NUMBER %OCTAVE_API_VERSION_NUMBER%
+
 #define OCTAVE_API_VERSION %OCTAVE_API_VERSION%
 
 #define OCTAVE_RELEASE_DATE %OCTAVE_RELEASE_DATE%
 
 #define OCTAVE_COPYRIGHT %OCTAVE_COPYRIGHT%
 
 // This is the first line printed by --version.  The GNU coding
 // standards say that the version number should follow the last space
